lib/hypgen/worker/runner.rb

```diff
@@ -22,7 +22,14 @@ def execute
           set_id = start_as(setup)
           workflow.set_set_id(set_id)
 
-          audit('Running workflow') { workflow.run! }
+          return_code = audit('Running workflow') { workflow.run! }
+          if return_code == 0
+            Hypgen.dap.update_exp(id, status: :finished)
+          else
+            Hypgen.dap.update_exp(id,
+                                  status: :error,
+                                  status_message: "Non zero return code!")
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.